smartstruct/lib/src/annotations.dart

```diff
@@ -17,3 +18,14 @@ class Mapping {
   final String target;
   const Mapping({required this.source, required this.target});
 }
+
+/// CustomMapping to support explicit field mapping in case the mapped target do not match in name,type or target is created based on few fields
+///
+/// Annotate the method with [CustomMapping] and provide a valid targetField and mapping function f
+/// To achive type safety, override CustomMapping class with specific TargetField and SourceClass types
+class CustomMapping<TargetField, SourceClass> {
```

Comment:
Right now it's not possible to support multiple source parameters in the custom mapping annotation.
This is currently not supported by the mapper anyway, but planned in the future (https://github.com/smotastic/smartstruct/issues/7)
so it would be nice if the customMapping would already support this (but not using it right now).

This also means, if we will have a dynamic amount of source parameters, we cannot define a static amount of parameters in the CustomMapping,
but dynamic is also not possible via generics.
So my first idea would be to not pass the custom mapping method as a parameter itself, but only the name of the method.
Something like
```dart
@Mapper()
abstract class CustomMapper {
  @CustomMapping(targetField: "fullName", f: 'mapFullName')
  Target fromSource(Source source);
}
```
The Mapper then could easily locate the method by it's name and just pass all source parameters which are available.
Alternatively we could define multiple CustomMapping Annotations like CustomMapping1, CustomMapping2, ..., which would then accept a multiple amount of source parameters,
but this sounds a little dirty and probably more complex to handle when we are reading them.